qiskit_experiments/calibration/calibration_definitions.py

```diff
@@ -0,0 +1,382 @@
+        """
+        Args:
+            name: The name of the schedule to get.
+            qubits: The qubits for which to get the schedule.
+            free_params: The parameters that should remain unassigned.
+            group: The calibration group from which to draw the
+                parameters. If not specifies this defaults to the 'default' group.
+
+        Returns:
+            schedule: A deep copy of the template schedule with all parameters assigned
+                except for those specified by free_params.
+
+        Raises:
+            CalibrationError: if the name of the schedule is not known.
+        """
+
+        # Get the schedule and deepcopy it to prevent binding from removing
+        # the parametric nature of the schedule.
+        if name not in self._schedules:
+            raise CalibrationError('Schedule %s is not defined.' % name)
+
+        sched = copy.deepcopy(self._schedules[name])
+
+        # Retrieve the channel indices based on the qubits and bind them.
+        binding_dict = {}
+        for ch in sched.channels:
+            if ch.is_parameterized():
+                index = self.get_channel_index(qubits, ch)
+                binding_dict[ch.index] = index
+
+        sched.assign_parameters(binding_dict)
+
+        # Loop through the remaining parameters in the schedule, get their values and bind.
+        if free_params is None:
+            free_params = []
+
+        binding_dict = {}
+        for inst in sched.instructions:
+            ch = inst[1].channel
+            for param in inst[1].parameters:
+                if param.name not in free_params:
+                    binding_dict[param] = self.parameter_value(param.name, ch, group=group)
+
+        sched.assign_parameters(binding_dict)
+
+        return sched
+
+    def get_circuit(self, name: str, qubits: Tuple, free_params: List[str] = None,
+                    group: Optional[str] = 'default', schedule: Schedule = None) -> QuantumCircuit:
+        """
+        Args:
+            name: The name of the gate to retrieve.
+            qubits: The qubits for which to generate the Gate.
+            free_params: Names of the parameters that will remain unassigned.
+            group: The calibration group from which to retrieve the calibrated values.
+                If unspecified this default to 'default'.
+            schedule: The schedule to add to the gate if the internally stored one is
+                not going to be used.
+
+        Returns:
+            A quantum circuit in which the parameter values have been assigned aside from
+            those explicitly specified in free_params.
+        """
+        if schedule is None:
+            schedule = self.get_schedule(name, qubits, free_params, group)
+
+        gate = Gate(name=name, num_qubits=len(qubits), params=list(schedule.parameters))
+        circ = QuantumCircuit(len(qubits), len(qubits))
```

Comment:
do we really need classical register?

Reply:
We do if we want to measure the qubits. Either we add it in all the `circuits` methods which seems redundant or we do it here. For now, I'm in favor of leaving this here to avoid duplicating code.